code/modules/mob/living/silicon/ai/ai_say.dm

```diff
@@ -1,11 +1,16 @@
-/mob/living/silicon/ai/compose_track_href(atom/movable/speaker, namepart)
+//the following 2 procs are staying here because its for AI and its shells (also AI controlled)
+/mob/living/silicon/compose_track_href(atom/movable/speaker, namepart)
+	if(!isAI(src) && !istype(src, /mob/living/silicon/robot/shell))
+		return ""
```

Comment:
Rather than doing `istype(src)`s, you should have a trait or something
`TRAIT_CAN_GET_AI_TRACKING_MESSAGE` or whatever.

Cause I'm sure someone's gonna wanna give this to, I dunno, the human ai shell or something at some point